[WIP] Add event creation page

AziizDev opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Introduced an event creation page with a comprehensive form for creating events in Arabic.
  - Implemented client-side form validation to ensure positive seat inputs.
  - Added a success modal for confirming event creation.

- **Improvements**
  - Enhanced page styling for better visual presentation and user experience.
  - Improved responsiveness and consistency across form controls.
  - Updated the `mongoose` dependency to the latest version for potential performance improvements.
  
- **Style**
  - Minor formatting adjustments made to enhance code readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

snyk-bot committed:
fix: upgrade mongoose from 8.8.3 to 8.9.1

Snyk has created this PR to upgrade mongoose from 8.8.3 to 8.9.1.

See this package in npm:
mongoose

See this project in Snyk:
https://app.snyk.io/org/abo3skr2019/project/b4b04c7e-d5c1-4541-84a5-85eb98820e5a?utm_source=github&utm_medium=referral&page=upgrade-pr

AziizDev committed:
feat(frontend): I've uploaded a css style for the event creation

AziizDev committed:
feat(frontend): I've uploaded a css style for the event creation

AziizDev committed:
feat(frontend): Added a Javascript file for the event creation page

AziizDev committed:
feat(frontend): Added a Javascript file for the event creation page

AziizDev committed:
feat(frontend): Added the EJS file for event creation page, with the links to important files in the head

AziizDev committed:
feat(frontend): Added the EJS file for event creation page, with the links to important files in the head

AziizDev committed:
feat(frontend): I've uploaded the EJS including the body for it with the partials

AziizDev committed:
feat(frontend): I've uploaded the EJS including the body for it with the partials

AziizDev committed:
feat(frontend): Now I've uploaded the sidebars for the page

AziizDev committed:
feat(frontend): Now I've uploaded the sidebars for the page

AziizDev committed:
feat(frontend): I've create the forms

AziizDev committed:
feat(frontend): I've create the forms

AziizDev committed:
feat(frontend): Pass current page context to event creation route

AziizDev committed:
feat(frontend): Pass current page context to event creation route

abo3skr2019 commented:
Remove the unneccessary edits to app.js 
as you already passed current page through the header.ejs 
and whenever possible use bootstrap instead of custom css 
here is an issue telling you why 
#54

AFAskar committed:
Merge pull request #39 from abo3skr2019/snyk-upgrade-595e15860100d297a1e5597d35501f8d

[Snyk] Upgrade mongoose from 8.8.3 to 8.9.1

snyk-bot committed:
fix: package.json & package-lock.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-MONGOOSE-8623536

SillyRobot883 committed:
feat(sidebar): integrate sidebar with event-creation page

SillyRobot883 commented:
I did some changes to the branch by integrated the new sidebar partial. Make sure to 
`git pull origin feature/frontend/event-creation` in order to get the new changes. 

Here's how it looks right now:
![image](https://github.com/user-attachments/assets/d3fce8a8-10d1-46b4-8d71-5da212f3052d)


> **Note:** the sidebar isn't merged to the dev branch _yet_, but when it's merged, do a `pull git` on dev to the sidebar files

AFAskar committed:
Merge pull request #55 from abo3skr2019/snyk-fix-60f0f5fd5f9387460710a0696a5ae094

AziizDev committed:
feat(frontend): I've deleted the css links to make bootstrap instead

AziizDev committed:
feat(frontend): make the ejs file used bootstrap

AziizDev committed:
removed the unneccessary changes on app.js

AziizDev committed:
resolve merge conflict

AziizDev committed:
feat(frontend): removed the unwanted things in the code

abo3skr2019 commented:
Remove Console logs and any other unnecessary js
and then move anything possible to bootstrap and i'll merge the PR

abo3skr2019 commented:
the poster should be a URL as currently we don't have file upload logic implemented , that will be in another phase as mentioned in #37
